Minor modifications to make compiling/using in Windows easier

GNSS-Stylist opened the pull request:
First: Thanks for this plugin! Works flawlessly once I got it working.

However, as I'm not very familiar with Godot's plugin/GDNative/GDExtension/module-infrastructure it took me a while to get this working. Thought that maybe the results from my experiments could be useful to someone else so I made this PR. Feel free to reject.

Also made a "release", mostly to keep track of my experiments: https://github.com/GNSS-Stylist/Godot-FFB-SDL/releases/tag/Dev1

Original commit comment:
To successfully compile this in windows a lib-file needs first to be generated out of the dll. This adds bat-scripts to do this. Also modified README.md to mention this Windows-related complication.

Added also "ffbplugin.gdextension"-file.

GNSS-Stylist committed:
Minor modifications to make compiling/using in Windows easier

To successfully compile this in windows a lib-file needs first to be generated out of the dll. This adds bat-scripts to do this. Also modified README.md to mention this Windows-related complication.

Added also "ffbplugin.gdextension"-file.

Dechode commented:
Hey, thanks for the pull request! Glad you got it working.

However i have been using mingw to cross-compile the binaries for windows from linux and the changes to Sconstruct script will brake this. I wonder if we can check for what compiler is being used?  

If so, I think it might be better to just grab the .lib file from the latest SDL2-devel-VC release from [SDL2 Releases here on github](https://github.com/libsdl-org/SDL/releases/tag/release-2.30.2) than to have multiple batch scripts for extracting it.

If its not possible then id like to keep cross-compiling an option because i don't have windows installed.

I will dive deeper tomorrow and and review the changes then. I'll also take a look at how we can set/check the compiler in SConstruct script

GNSS-Stylist committed:
Minor change to SConstruct to allow compilation with MinGW

Do MinGW and VC handle paths differently?

GNSS-Stylist commented:
Investigated this a bit and made a minor fix to SConstruct to allow compilation with MinGW under windows. Compilation goes through, but the resulting dll doesn't, however, seem to work. First Godot nags about missing dependencies when trying to use it:
> Can't open dynamic library: C:/long_path/Godot-FFB-SDL-Extension/bin/libffbplugin.windows.template_debug.x86_64.dll. Missing dependencies: cygwin1.dll, cyggcc_s-seh-1.dll, cygstdc++-6.dll. Error: Error 126: The specified module could not be found..
> core/extension/gdextension.cpp:455 - GDExtension dynamic library not found: C:/long_path/Godot-FFB-SDL-Extension/bin/libffbplugin.windows.template_debug.x86_64.dll

After adding C:\cygwin64\bin to path Godot nags about the dll itself missing, which sounds a bit misleading:
> Can't open dynamic library: C:/long_path/Godot-FFB-SDL-Extension/bin/libffbplugin.windows.template_debug.x86_64.dll. Error: Error 126: The specified module could not be found..

Not sure what's going on here, but will happily continue using the working dll compiled with VC.

Grabbing the lib-file from SDL2-releases also work, thanks for the tip!

Dechode commented on generate_SDL2_lib_for_windows.bat:
Now that you got the plugin compiling with .lib files grabbed from sdl releases github page, you can remove these .bat scripts.

GNSS-Stylist replied:
Removed

Dechode commented on SConstruct:
The original lib include is needed for directly linking with the dll when compiling with MinGW. SCons seems to strip the .dll suffix from the lib path, but only the first one. Hence the weird 'SDL2.dll.dll' path name.

This is why we should check which compiler is being used and append the appropriate path. I found that you can use:
`
env['CXX']
`
inside the SConstruct script to see what compiler is being used.

GNSS-Stylist replied:
Added separate library definition for Visual C++ and added SDL2_VC.lib-file downloaded from SDL2-releases. Now compiles in windows/Visual C++ without problems. Does not compile with MinGW ("cannot find -lSDL2.dll.dll: No such file or directory"), but not sure if this could be fixed or worth trying(?)

GNSS-Stylist committed:
Add SDL2_VC.lib for Microsoft Visual C++ and update SDL2.dll

New SDL2.dll version 2.30.2. Tested in Windows 10 with Logitech Driving Force GT.

GNSS-Stylist committed:
Remove unnecessary scripts, update SConstruct and README.md

Scripts were used to generate .lib-file from SDL2.dll to make the plugin compilable with Visual C++. The .lib-file can, however be downloaded from SDL2's releases (and is now included in the repo), so there's actually no need for this.

No need to mention the aforementioned scripts nor Visual C++-related intricacies in README.md

SConstruct now uses different .lib when compiling with Visual C++

GNSS-Stylist commented:
Made some commits and now the plugin compiles with VC without complications. Also updated the SDL2.dll to latest stable V2.30.2 (because why not if downloading the corresponding .lib-file anyway). Tested with Win 10 and Logitech Driving Force GT.

Dechode commented on SConstruct:
```diff
@@ -17,7 +17,13 @@ env = SConscript("godot-cpp/SConstruct")
     env.Append(CPPPATH=['include/'])
     env.Append(LIBPATH=['lib/SDL2/'])
-    env.Append(LIBS=['SDL2.dll.dll'])
+    print("CXX is: " + env["CXX"])
+    if env["CXX"] == "$CC":
```
Does "$CC" represent the msvc compiler here? Just curious cause for mingw it was a lot longer, something like "mingw32_64_something"

GNSS-Stylist replied:
I just tested what the print-statement in line 20 printed and used it. It prints "CXX is: g++" when compiling with minGW in windows Command Prompt (command "scons use_mingw=true") and "CXX is: $CC" when using Visual C++ (command just "scons").

godot-cpp/SConstruct actually seems to determine the compiler another way (line 236+):
```
# Require C++17
if env.get("is_msvc", False):
    env.Append(CXXFLAGS=["/std:c++17"])
else:
    env.Append(CXXFLAGS=["-std=c++17"])
```

Tested this and it seems to work in the place of the previous check. I can commit this change if you like(?) "is_msvc" feels  more self-descriptive.

Dechode reviewed (commented):
Looking good now. I will still try the changes myself once i have the time.
